float32 rightVelocity
float32 leftVelocity